### PR TITLE
Addition of new resource and data source for L4-L7 Concrete Device(vnsCDev)

### PR DESCRIPTION
**Pull request by shrsr**

### CODE REVIEW

File: website/docs/r/concrete_device.html.markdown [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  vcenter_name                     = "vcenter"
+  vm_name                          = "tenant1-ASA1"
+  relation_vns_rs_c_dev_to_ctrlr_p = "uni/vmmp-VMware/dom-ACI-vDS/ctrlr-vcenter"
+}
+```
+
+## Argument Reference ##
+
+* `l4-l7_devices_dn` - (Required) Distinguished name of the parent L4-L7 Devices object.
+* `name` - (Required) Name of the object Concrete Device.
```
Comment (lhercot) [inline website/docs/r/concrete_device.html.markdown:37]: of the Concrete Device object

File: website/docs/r/concrete_device.html.markdown [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+* `annotation` - (Optional) Annotation of the object Concrete Device.
```
Comment (lhercot) [inline website/docs/r/concrete_device.html.markdown:38]: of the Concrete Device object

File: website/docs/r/concrete_device.html.markdown [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+* `name_alias` - (Optional) Name Alias of object Concrete Device.
+* `vcenter_name` - (Optional) The virtual center name on which the device is hosted in the L4-L7 device cluster. It uniquely identifies the virtual center. Type: String.
```
Comment (lhercot) [inline website/docs/r/concrete_device.html.markdown:40]: The vCenter name on ...
It uniquely identifies the vCenter.

Comment (lhercot) [reply website/docs/r/concrete_device.html.markdown:40]: Is this actually used by the UI?

File: website/docs/r/concrete_device.html.markdown [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+* `vm_name` - (Optional) The virtual center VM name on which the device is hosted in the L4-L7 device cluster. It uniquely identifies the VM. Type: String.
```
Comment (lhercot) [inline website/docs/r/concrete_device.html.markdown:41]: The name of the Virtual Machine (VM) in the vCenter on which ...

File: website/docs/d/concrete_device.html.markdown [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+* `l4-l7_devices_dn` - (Required) Distinguished name of parent L4-L7 Devices object.
+* `name` - (Required) Name of object Concrete Device.
+
+## Attribute Reference ##
+* `id` - Attribute id set to the Dn of the Concrete Device.
+* `annotation` - (Optional) Annotation of object Concrete Device.
+* `name_alias` - (Optional) Name Alias of object Concrete Device.
+* `vcenter_name` - (Optional) The virtual center name on which the device is hosted in the L4-L7 device cluster. It uniquely identifies the virtual center. Type: String.
+* `vm_name` - (Optional) The virtual center VM name on which the device is hosted in the L4-L7 device cluster. It uniquely identifies the VM. Type: String.
+* `relation_vns_rs_c_dev_to_ctrlr_p` - (Optional) Represents the relation from a Concrete Device to a VMM Controller (class vmmCtrlrP). It is an implicit relation to validate the controller profile. Type: String.
```
Comment (lhercot) [inline website/docs/d/concrete_device.html.markdown:42]: What does "It is an implicit relation to validate the controller profile." mean?

File: website/docs/r/concrete_device.html.markdown [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+* `relation_vns_rs_c_dev_to_ctrlr_p` - (Optional) Represents the relation from a Concrete Device to a VMM Controller (class vmmCtrlrP). It is an implicit relation to validate the controller profile. Type: String.
```
Comment (lhercot) [inline website/docs/r/concrete_device.html.markdown:42]: What does "It is an implicit relation to validate the controller profile." mean?

**Review by lhercot — CHANGES_REQUESTED** [review]

File: website/docs/r/concrete_device.html.markdown (hunk 3 above)
Comment (shrsr) [reply website/docs/r/concrete_device.html.markdown:40]: Yes, it is on the UI but is not configurable there.

File: website/docs/d/concrete_device.html.markdown [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+## Argument Reference ##
+
+* `l4-l7_devices_dn` - (Required) Distinguished name of parent L4-L7 Devices object.
+* `name` - (Required) Name of object Concrete Device.
+
+## Attribute Reference ##
+* `id` - Attribute id set to the Dn of the Concrete Device.
+* `annotation` - (Optional) Annotation of object Concrete Device.
+* `name_alias` - (Optional) Name Alias of object Concrete Device.
+* `vcenter_name` - (Optional) The virtual center name on which the device is hosted in the L4-L7 device cluster. It uniquely identifies the virtual center. Type: String.
```
[the remark this answers is not in the document]
Comment (shrsr) [reply website/docs/d/concrete_device.html.markdown:40]: Removed

File: website/docs/r/concrete_device.html.markdown (hunk 6 above)
Comment (shrsr) [reply website/docs/r/concrete_device.html.markdown:42]: Removed

File: website/docs/d/concrete_device.html.markdown (hunk 5 above)
Comment (shrsr) [reply website/docs/d/concrete_device.html.markdown:42]: Removed

File: examples/l4_l7_concrete_device/main.tf [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    domain_dn = "uni/vmmp-VMware/dom-ACI-vDS"
+  }
+}
+
+resource "aci_concrete_device" "example" {
+  l4_l7_device_dn   = aci_l4_l7_device.example.id
+  name              = "example"
+  vmm_controller_dn = "uni/vmmp-VMware/dom-ACI-vDS/ctrlr-vcenter"
+  vm_name           = "tenant1-ASA1"
+}
```
Comment (lhercot) [inline examples/l4_l7_concrete_device/main.tf:43]: Add an example for Physical device.

File: website/docs/d/concrete_device.html.markdown [hunk 9]
```diff
@@ -0,0 +1,41 @@
+---
+layout: "aci"
```
Comment (lhercot) [inline website/docs/d/concrete_device.html.markdown:2]: Add subcategory

File: website/docs/r/concrete_device.html.markdown [hunk 10]
```diff
@@ -0,0 +1,49 @@
+---
+layout: "aci"
```
Comment (lhercot) [inline website/docs/r/concrete_device.html.markdown:2]: Add subcategory

File: website/docs/r/concrete_device.html.markdown [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+
+## Example Usage ##
+
+```hcl
+resource "aci_concrete_device" "example" {
+  l4_l7_device_dn   = aci_l4_l7_device.example.id
+  name              = "example"
+  vmm_controller_dn = "uni/vmmp-VMware/dom-ACI-vDS/ctrlr-vcenter"
+  vm_name           = "tenant1-ASA1"
+}
```
Comment (lhercot) [inline website/docs/r/concrete_device.html.markdown:30]: Add also an example for physical

File: website/docs/r/concrete_device.html.markdown [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+```
+
+## Argument Reference ##
+
+* `l4-l7_device_dn` - (Required) Distinguished name of the parent L4-L7 Device object.
+* `name` - (Required) Name of the Concrete Device object.
+* `annotation` - (Optional) Annotation of the Concrete Device object.
+* `name_alias` - (Optional) Name Alias of the Concrete Device object.
+* `vmm_controller_dn` - (Optional) Distinguished name of the VMM controller object. Type: String.
```
Comment (lhercot) [inline website/docs/r/concrete_device.html.markdown:39]: Add that this is used for Virtual L4-L7 Devices only

File: website/docs/r/concrete_device.html.markdown [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+* `vm_name` - (Optional) The name of the Virtual Machine (VM) in the vCenter on which the device in the L4-L7 device cluster is hosted. It uniquely identifies the VM. Type: String.
```
Comment (lhercot) [inline website/docs/r/concrete_device.html.markdown:40]: Add that this is used for Virtual L4-L7 Devices only

**Review by lhercot — CHANGES_REQUESTED** [review]

File: examples/l4_l7_concrete_device/main.tf [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+  l4_l7_device_dn   = aci_l4_l7_device.example.id
+  name              = "virtual-Device"
+  vmm_controller_dn = "uni/vmmp-VMware/dom-ACI-vDS/ctrlr-vcenter"
+  vm_name           = "tenant1-ASA1"
+}
+
+resource "aci_concrete_device" "example2" {
+  l4_l7_device_dn   = aci_l4_l7_device.example.id
+  name              = "physical-Device"
+}
```
Comment (lhercot) [inline examples/l4_l7_concrete_device/main.tf:48]: Create a separate Physical l4_l7_device for this one.

**Review by lhercot — CHANGES_REQUESTED** [review]

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**